src/Controller/InAppPurchasesController.php

```diff
@@ -0,0 +1,138 @@
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Store\InAppPurchase\Services\InAppPurchaseUpdater;
+use Shopware\Core\Framework\Store\Services\AbstractExtensionDataProvider;
+use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use SwagExtensionStore\Exception\ExtensionStoreException;
+use SwagExtensionStore\Services\InAppPurchasesService;
+use SwagExtensionStore\Struct\InAppPurchaseCartPositionCollection;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Attribute\Route;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+#[Route(defaults: ['_routeScope' => ['api']])]
+class InAppPurchasesController
+{
+    /**
+     * @param EntityRepository<AppCollection> $appRepository
+     */
+    public function __construct(
+        private readonly InAppPurchasesService $inAppPurchasesService,
+        private readonly InAppPurchaseUpdater $inAppPurchaseUpdater,
+        private readonly AbstractExtensionDataProvider $extensionDataProvider,
+        private readonly InAppPurchasesGateway $appPurchasesGateway,
+        private readonly EntityRepository $appRepository,
+    ) {}
+
+    #[Route('/api/_action/in-app-purchases/{technicalName}/details', name: 'api.in-app-purchases.detail', methods: ['GET'])]
+    public function getInAppFeature(string $technicalName, Context $context): Response
+    {
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('name', $technicalName));
+
+        $extension = $this->extensionDataProvider
+            ->getInstalledExtensions($context, false, $criteria)
+            ->first();
+
+        return new JsonResponse($extension);
+    }
+
+    #[Route('/api/_action/in-app-purchases/cart/new', name: 'api.in-app-purchases.cart.new', methods: ['POST'])]
+    public function createCart(RequestDataBag $data, Context $context): Response
+    {
+        $name = $data->getString('name');
+        $feature = $data->getString('feature');
+
+        $cart = $this->inAppPurchasesService->createCart($name, $feature, $context);
+
+        return new JsonResponse($cart);
+    }
+
+    #[Route('/api/_action/in-app-purchases/cart/order', name: 'api.in-app-purchases.cart.order', methods: ['POST'])]
+    public function orderCart(RequestDataBag $data, Context $context): Response
+    {
+        $taxRate = \floatval($data->getString('taxRate'));
+        $positions = $data->get('positions');
+        $extensionName = $data->get('name');
+
+        $positionCollection = InAppPurchaseCartPositionCollection::fromArray($positions->all());
+
+        $app = $this->getAppByName($extensionName, $context);
+        if (!$app) {
+            return $this->inAppPurchasesService->orderCart($taxRate, $positionCollection->toCart(), $context);
```

Comment:
if the app is not installed, we just do the purchase? 🤔

Reply:
If it's not an app, it's a plugin, and then we process the checkout. Do you want me to put in a plugin check here too?

Reply:
no, I just didn't realize that this was about the gateway, all good.